src/azure-cli/azure/cli/command_modules/resource/_validators.py

```diff
@@ -16,6 +16,48 @@
 MSI_LOCAL_ID = '[system]'
 
 
+def _validate_template_input(namespace):
+    if namespace.template_file and not os.path.isfile(namespace.template_file):
+        raise CLIError('Please enter a valid file path')
+
+
+def _validate_template_spec(namespace):
+    if namespace.template_spec is None:
+        if (namespace.name is None or namespace.resource_group_name is None):
+            raise CLIError('incorrect usage: Please enter '
+                           'a resource group and resource name or a resource ID for --template-spec')
+    else:
+        from msrestazure.tools import is_valid_resource_id
+        if not is_valid_resource_id(namespace.template_spec):
+            raise CLIError('--template-spec is not a valid resource ID.')
+
+
+def _validate_template_spec_out(namespace):
+    _validate_template_spec(namespace)
+    if namespace.output_folder and not os.path.isdir(namespace.output_folder):
+        raise CLIError('Please enter a valid output folder')
+
+
+def _validate_deployment_name_with_template_specs(namespace):
+    # If missing,try come out with a name associated with the template name
+    if namespace.deployment_name is None:
+        template_filename = None
+        if namespace.template_file and os.path.isfile(namespace.template_file):
+            template_filename = namespace.template_file
+        if namespace.template_uri and urlparse(namespace.template_uri).scheme:
+            template_filename = urlsplit(namespace.template_uri).path
+        if namespace.template_spec:
+            from msrestazure.tools import parse_resource_id, is_valid_resource_id
```

Comment:
Use "from azure.mgmt.core.tools import parse_resource_id,  is_valid_resource_id" as we'll deprecate "msrestazure.tools" in the future.